Makefile

```diff
@@ -18,56 +16,58 @@ all: help
 
 .PHONY: help
 help:
-	@cat README.md
-
-.PHONY: run
-run: bin/python setup.py
-	PYTHONPATH=src bash -c 'bin/python src/metaswitch/crest/main.py'
+	@cat docs/development.md
 
 .PHONY: test
-test: bin/python setup.py
-	bin/python setup.py test
+test: setup.py env
+	PYTHONPATH=src:common ${ENV_DIR}/bin/python setup.py test -v
+
+${ENV_DIR}/bin/flake8: env
+	${ENV_DIR}/bin/pip install flake8
+
+${ENV_DIR}/bin/coverage: env
+	${ENV_DIR}/bin/pip install coverage
 
-verify:
-	flake8 --select=E10,E11,E9,F src/
+verify: ${ENV_DIR}/bin/flake8
+	${ENV_DIR}/bin/flake8 --select=E10,E11,E9,F src/
 
-style:
-	flake8 --select=E,W,C,N --max-line-length=${MAX_LINE_LENGTH} src/
+style: ${ENV_DIR}/bin/flake8
+	${ENV_DIR}/bin/flake8 --select=E,W,C,N --max-line-length=100 src/
 
-explain-style:
-	flake8 --select=E,W,C,N --show-pep8 --first --max-line-length=${MAX_LINE_LENGTH} src/
+explain-style: ${ENV_DIR}/bin/flake8
+	${ENV_DIR}/bin/flake8 --select=E,W,C,N --show-pep8 --first --max-line-length=100 src/
 
 .PHONY: coverage
-coverage: bin/coverage setup.py
-	# Explictly force use of bin/python so we load the correct python modules
+coverage: ${ENV_DIR}/bin/coverage setup.py
 	rm -rf htmlcov/
-	bin/python bin/coverage erase
-	bin/python bin/coverage run --source src --omit "src/metaswitch/**/test/**"  setup.py test
-	bin/python bin/coverage report -m
-	bin/python bin/coverage html
+	${ENV_DIR}/bin/coverage erase
+	${ENV_DIR}/bin/coverage run --source src --omit "**/test/**"  setup.py test
+	${ENV_DIR}/bin/coverage report -m
+	${ENV_DIR}/bin/coverage html
 
 .PHONY: env
-env: bin/python
-
-bin/python bin/coverage: bin/buildout buildout.cfg
-ifeq ($(X86_64_ONLY),1)
-	ARCHFLAGS="-arch x86_64" ./bin/buildout -N
-else
-	ARCHFLAGS="-arch i386 -arch x86_64" ./bin/buildout -N
-endif
-	${ENV_DIR}/bin/easy_install -zmaxd eggs/ zc.buildout
-
-bin/buildout: $(ENV_DIR)/bin/python
-	mkdir -p .buildout_downloads/dist
-	cp thrift_download/thrift-0.8.0.tar.gz .buildout_downloads/dist/
+env: ${ENV_DIR}/.eggs_installed
+
+$(ENV_DIR)/bin/python: setup.py common/setup.py
+	# Set up a fresh virtual environment
+	virtualenv --setuptools --python=$(PYTHON_BIN) $(ENV_DIR)
 	$(ENV_DIR)/bin/easy_install "setuptools>0.7"
 	$(ENV_DIR)/bin/easy_install distribute
-	$(ENV_DIR)/bin/easy_install zc.buildout
-	mkdir -p bin/
-	ln -s $(ENV_DIR)/bin/buildout bin/
-
-$(ENV_DIR)/bin/python:
-	virtualenv --no-site-packages --setuptools --python=$(PYTHON_BIN) $(ENV_DIR)
+	
+${ENV_DIR}/.eggs_installed : $(ENV_DIR)/bin/python $(shell find src/metaswitch -type f -not -name "*.pyc") $(shell find common/metaswitch -type f -not -name "*.pyc")
+	# Generate .egg files for crest and python-common
+	${ENV_DIR}/bin/python setup.py bdist_egg -d .eggs
+	cd common && EGG_DIR=../.eggs make build_common_egg
```

Comment:
Building python-common requires appropriate compiler flags (to pull in cpp-common for cffi) - I've wrapped that ion a Makefile target.